pkgs/development/tools/go-task/default.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, fetchurl }:
+
+let
+  targetPath = "$out/bin";
+
+in
+stdenv.mkDerivation {
+  pname   = "go-task";
+  version = "2.8.1";
+
+  src = fetchurl {
+    url    = "https://github.com/go-task/task/releases/download/v2.8.1/task_linux_amd64.tar.gz";
+    sha256 = "1fhshi5y7lg479b2lg02swc7alwwgvn1gm4h9yq2bnx6hppnkjn7";
+  };
```

Comment:
Compiling from source is strongly preferred in nixpkgs. Since this looks like a regular Go project, you could use `buildGoModule` (or `buildGoPackage`).

If you build from source, since the project is on GitHub, you can use `fetchFromGitHub` in place of `fetchurl`.

Reply:
Yeah that makes total sense. I am a bit confused about how to go about this for this particular package, as it leverages [GoReleaser](https://goreleaser.com/#continuous_integration) for its build. I will try to find some time to dig into it and see if I can't get it working with `buildGoPackage` instead. Thanks!

Reply:
I tried with `buildGoModule` first, as it does seem to be a go module, but it was unsuccessful in building `subPackages`. I tried to do it with `buildGoPackage` instead (leveraging `vgo2nix`), and it seems like it has mostly worked, but one thing I've noticed is when running `task --version` it produces `Task version: master` instead of the actual version of the tool. From looking into it a little, it seems like `GoReleaser` pulls the version from the most recent tag and injects it somehow into the build. I'm not terribly familiar with golang packages unfortunately and I haven't been able to determine how it does that, or how to replicate that behavior.

Reply:
The version is `cmd/task/task.go`: https://github.com/go-task/task/blob/86be13ff1f1173479d1c032bc1b9eae67f06f64e/cmd/task/task.go#L19

You should be able to use `substituteInPlace` to replace that by an actual version in `postPatch`. E.g. (untested):

```
  postPatch = ''
    substituteInPlace cmd/task/task.go \
      'version = "master"' 'version = "${version}"'
  '';
```

Reply:
Thanks! That indeed did do the trick!